### ISSUE TITLE
Have stories on each level of hierarchy

### ISSUE THREAD

**Issue by igor-dv** [storybooks/storybook#1699]
This is very asked improvement from the beginning of the hierarchy feature.

Today we have this:
```js
storiesOf('Cells/Molecules', module)
  .add('with text', () => <Button>Hello Button</Button>)
  .add('with some emoji', () => <Button>😀 😎 👍 💯</Button>);

storiesOf('Cells/Molecules/Atoms', module)
  .add('with text2', () => <Button>Hello Button</Button>)
  .add('with some emoji2', () => <Button>😀 😎 👍 💯</Button>);
```

That will be shown as 
```
- Cells
  - Molecules                
    - with text                
    - with some emoji   
  - Molecules               
    - Atoms                   
      - with text 2
      - with some emoji 2
```

Desired
```
- Cells
  - Molecules                
    - with text                
    - with some emoji   
    - Atoms                   
      - with text 2
      - with some emoji 2
```

In this case we should allow stories and groups to be in the same level.

**Comment by igor-dv** [comment]
I would like us to discuss this. Do we want to implement this ? Do we want to keep it configurable ?

For example the first thing I can think about is to remove the story auto-selection  when clicking the storyKind.

**Comment by TomiS** [comment]
Yes! I ran into this problem in less than 10 minutes with Storybook. I'd also consider this as a bug and not a feature improvement. I think Storybook should merge the nested structure independent of how it is exactly defined.

**Comment by ndelangen** [comment]
I think the desired output IS clearly what everyone expects and wants us to do.

And I agree 100% that's what we want too.
Who's interesting in making a PR for this?

**Comment by igor-dv** [comment]
I'll take it 😉

**Comment by igor-dv** [comment]
What do you think about always putting stories at the beginning ? ()

![image](https://user-images.githubusercontent.com/7867954/29818340-942fd8e2-8cc4-11e7-8971-a3b6b7444f2c.png)

**Comment by ndelangen** [comment]
@igor-dv I'm pretty sure that's what everyone is expecting ✅

**Comment by ndelangen** [comment]
This feature is complete and merged to the 3.3 branch
